source/deployment/mobile-app-deployment.rst

```diff
@@ -0,0 +1,571 @@
+
+Custom Apps
+===========
+
+If you desire to customize the applications, or do not want your users downloading the application from the public app stores, you will need to build the apps yourself. 
+
+As the apps are an open source project, it will require a fork, and your team will be responsible for maintaining the fork, as well as keeping the fork up to date with any changes made by Mattermost.
+
+This process can be complicated. It will also greatly increase deployment time, not only initially, but whenever the apps need to be updated. 
+
+We recommend having your development team take a look at the documentation to ensure they understand the scale and requirements of this path. In general, this route will present some challenges, including…
+
+- Obtaining/providing certificates for your custom application
+- Signing your custom applications
+- Distributing your applications via public or private app stores
+
+*******************************************
+Mattermost Push Notification Service (MPNS)
+*******************************************
+
+Receiving notifications on a mobile device is a core value of your mobile deployment.
+
+It also represents a return on your deployment investment through better-connected users. 
+
+A push proxy is a key technology behind notification transmission.  It enables notifications between the server and the mobile application.
+
+Mattermost provides a self-hosted Push Proxy you can deploy, the Mattermost Push Notification Service (MPNS). This is also available via a hosted-by-mattermost option, or hosted push notification service (HPNS)*.
+
+    **Note:** `Only Mattermost Enterprise Edition E10 <https://about.mattermost.com/pricing/>`_ or higher to receive access to our Hosted Push Notification Service (HPNS)
+
+If you will be using the Mattermost applications via the iOS and Android app stores, the HPNS is all you need.
+
+Our Hosted Push Notification Service offers:
+
+- Access to a publicly-hosted Mattermost Push Notification Service (MPNS).
+- An explicit `privacy policy <https://about.mattermost.com/hpns-privacy/>`_ for the contents of unencrypted messages.
+- Encrypted TLS connections:
+    - between HPNS and Apple Push Notification Services
+    - HPNS and Google’s Firebase Cloud Messaging service
+    - HPNS and your Mattermost server
+- Production-level uptime expectations
+- Compatibility with EMM Providers*
+
+\* *When using our publicly available app store applications and the AppConfig standard.*
+
+Of course, if you would like to host your own version of the MPNS, that is always an option. The `Choosing the Right Deployment Model <https://docs.google.com/document/d/1BhHla4medUGVSWuBPNuoz_h5f85MdlIQWcRn5N2WUos/edit#heading=h.6zz62c4mus3l>`_ section of this guide, as well as our `Mobile App Admin Documentation <https://docs.mattermost.com/mobile/mobile-hpns.html>`_ are the best places to start.
```

Comment:
Do we need to mention that hosting your own MPNS requires a custom build?

Reply:
Again, this was an effort to make a push towards a much more efficient path.

Reply:
+1. I missunderstod. We should call this out as often as possible.

Reply:
Suggestion: `Of course, if you would like to host your own version of the MPNS, that is always an option but requires a custom built app.`

Reply:
Per @enahum the only change I would make is to add, "..., but this will require you to build the mobile apps yourself." - or something similar.

Reply:
resolved in meeting